src/devtran.cpp

```diff
@@ -518,14 +517,16 @@ Rcpp::List DEVTRAN(const Rcpp::List parin,
       if(this_rec->is_event()) {
         
         this_cmtn = this_rec->cmtn();
+
+        if(!this_rec->is_lagged()) {
```

Comment:
This is key ... if this is a lagged dose (a phantom record) we _cannot_ change bioavailability here.